SurveyTests/Sources/Specs/Presentation/Modules/SurveyQuestion/SurveyQuestionViewModelSpec.swift

```diff
@@ -0,0 +1,185 @@
+        var input: SurveyQuestionViewModel.Input!
+        var output: SurveyQuestionViewModel.Output!
+        var viewModel: SurveyQuestionViewModel!
+        let submitTrigger = PassthroughSubject<Void, Never>()
+        let dismissAlert = PassthroughSubject<Void, Never>()
+        let clearSubmissionTrigger = PassthroughSubject<Void, Never>()
+        let onAppearTrigger = PassthroughSubject<[SurveyQuestion], Never>()
+
+        describe("A SurveyQuestionViewModel") {
+
+            let id = "d5de6a8f8f5f1cfe51bc"
+
+            beforeEach {
+                submitSurveyUseCase = SubmitSurveyUseCaseMock()
+                submissionStorageUseCase = SubmissionStorageUseCaseMock()
+                viewModel = SurveyQuestionViewModel(
+                    submitSurveyUseCase: submitSurveyUseCase,
+                    submissionStorageUseCase: submissionStorageUseCase,
+                    id: id
+                )
+
+                input = SurveyQuestionViewModel.Input(
+                    submitTrigger: submitTrigger.eraseToAnyPublisher(),
+                    dismissAlert: dismissAlert.eraseToAnyPublisher(),
+                    clearSubmissionTrigger: clearSubmissionTrigger.eraseToAnyPublisher(),
+                    onAppearTrigger: onAppearTrigger.eraseToAnyPublisher()
+                )
+                output = viewModel.transform(input)
+            }
+
+            describe("its onAppearTrigger called") {
+
+                beforeEach {
+                    onAppearTrigger.send(APISurveyQuestion.dummy)
+                }
+
+                context("when submissionStorageUseCase calls store") {
+
+                    it("returns output done") {
+                        expect(output.done) == ()
+                    }
+                }
+
+                context("when submissionStorageUseCase calls delete") {
+
+                    it("submissionStorageUseCase calls delete") {
+                        expect(submissionStorageUseCase.deleteCalled) == true
+                    }
+
+                    it("returns correct response") {
+                        _ = submissionStorageUseCase.delete()
+                            .sink { _ in
+                            } receiveValue: {
+                                expect($0) == true
+                            }
+                    }
+                }
+            }
+
+            describe("its clearSubmissionTrigger called") {
+
+                beforeEach {
+                    clearSubmissionTrigger.send()
+                }
+
+                context("when submissionStorageUseCase calls delete") {
+
+                    it("returns output done") {
+                        expect(output.done) == ()
+                    }
+
+                    it("returns a correct response") {
+                        _ = submissionStorageUseCase.delete()
+                            .sink { _ in
+                            } receiveValue: {
+                                expect($0) == true
```

Comment:
Somehow all of these async calls are picked up by XCTest. Normally we have to do it asynchronously or the test will be ignored, like in this [article](https://www.swiftbysundell.com/articles/unit-testing-combine-based-swift-code/). Not sure if the mock is so fast or some logic are sync underneath 🤔

Reply:
Although we're using Combine, the test is like the traditional synchronous way. Consider applying the [article](https://www.swiftbysundell.com/articles/unit-testing-combine-based-swift-code/) @blyscuit mentioned above; I also apply this in my IC.

Reply:
Thank you for the suggestion. It's a nice article! I fixed all the comments and applied `awaitPublisher` to Unit tests at https://github.com/nkhanh44/ic-survey-ios/pull/88/commits/97b7c269f7a799da63d092c053475bc3eee32b1b, I also updated the QuestionSubmissionStorage at https://github.com/nkhanh44/ic-survey-ios/pull/88/commits/18ba1fd38742a5d87f46e36fbae4fbfac4c60484.